frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx

```diff
@@ -0,0 +1,404 @@
+                isSubmitting,
+                formContext,
+              }}
+            />
+          ))}
+        </VStack>
+        {schema.allowMultiple ? (
+          <HStack>
+            <Button
+              isDisabled={isSubmitting}
+              alignSelf="start"
+              leftIcon={<BiPlus />}
+              aria-label="Add another child"
+              onClick={() => {
+                append([''])
+              }}
+            >
+              Add another child
+            </Button>
+            <Spacer />
+            <Text
+              textStyle="body-2"
+              color="secondary.400"
+            >{simplur`${numChild} child[|ren] added`}</Text>
+          </HStack>
+        ) : null}
+      </VStack>
+    </FieldContainer>
+  )
+}
+
+interface ChildrenBodyProps {
+  currChildBodyIdx: number
+  schema: ChildrenCompoundFieldSchema
+  fields: FieldArrayWithId<
+    ChildrenCompoundFieldInputs,
+    `${string}.child`,
+    'id'
+  >[]
+  field: FieldArrayWithId<ChildrenCompoundFieldInputs, `${string}.child`, 'id'>
+  colorTheme: FormColorTheme
+  remove: UseFieldArrayRemove
+  myInfoChildrenBirthRecords?: MyInfoChildData
+  isSubmitting: boolean
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  formContext: UseFormReturn<ChildrenCompoundFieldInputs, any>
+}
+
+const ChildrenBody = ({
+  currChildBodyIdx,
+  schema,
+  fields,
+  field,
+  colorTheme,
+  remove,
+  myInfoChildrenBirthRecords,
+  isSubmitting,
+  formContext,
+}: ChildrenBodyProps): JSX.Element => {
+  const { register, getValues, setValue, watch } = formContext
+
+  const childNamePath = useMemo(
+    () => `${schema._id}.child.${currChildBodyIdx}.0`,
+    [schema._id, currChildBodyIdx],
+  )
+  const { onChange: selectOnChange, ...selectRest } = register(childNamePath)
+
+  const childName = watch(childNamePath) as unknown as string
+
+  const allChilds = useMemo<string[]>(() => {
+    if (myInfoChildrenBirthRecords === undefined) {
+      return []
+    }
+    return myInfoChildrenBirthRecords[MyInfoAttribute.ChildName] ?? []
+  }, [myInfoChildrenBirthRecords])
+
+  // useCallback to re-compute names because for some reason watch doesn't
+  // work on this nested field in react-hook-form.
+  const allSelectedNames = useCallback((): string[] => {
+    // Really important to note that sometimes react-hook-form stores our "array"
+    // as a object with key=index and values=array entry.
+    return Object.values(getValues(`${schema._id}.child`)).map((arr) => arr[0])
+  }, [getValues, schema._id])
+
+  // useCallback to re-compute names, again because of buggy allSelectedNames
+  const namesNotSelected = useCallback((): string[] => {
+    if (myInfoChildrenBirthRecords === undefined) {
+      return []
+    }
+    const temp = allSelectedNames()
+    // We want all child names that haven't already been selected.
+    // O(n^2) but n is small so it should be okay.
+    return allChilds.filter((name) => !temp.includes(name))
```

Comment:
👍 

just for a note, you could use a `Set` here here, but for small arrays, it's probably slower than filtering with array.includes()` 🤷 

```suggestion
    const temp = new Set(allSelectedNames())
    // We want all child names that haven't already been selected.
    return allChilds.filter((name) => !temp.has(name))
```

Reply:
> Valentina Vassilyeva and her husband Feodor Vassilyev are alleged to hold the record for the most children a couple has produced. She gave birth to a total of 69 children

-Wikipedia.

Guess I will just use set for such cases. Though you're right it's probably slower for almost all cases.